### ISSUE TITLE
Rails/Date detects strings as Date objects

### ISSUE THREAD

**Issue by jcoyne** [rubocop-hq/rubocop#6438]
When I have this code:
```ruby
      value = '2001-10-20T10:22:22Z'
      value.to_time
```

then rubycop reports this fault:
```
C: Rails/Date: Do not use to_time on Date objects, because they know nothing about the time zone in use.
      value.to_time
            ^^^^^^^
``` 
--------

## Expected behavior
Clearly that's not a date object, it's a string object.  It should not insist that it is a Date.

## Actual behavior

It says a String is a Date.

## Steps to reproduce the problem

```ruby
      value = '2001-10-20T10:22:22Z'
      value.to_time
```

## RuboCop version

```
$ bundle exec rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by Drenmi** [comment]
Unfortunately runtime type inference is outside the scope of what we can take on with static analysis. The case where `#to_time` is sent directly to a string literal, e.g.:

```ruby
'2001-10-20T10:22:22Z'.to_time
```

is already handled as a special case by the cop.

**Comment by jcoyne** [comment]
@Drenmi if that is the case, then why does Rails/Date use type inferencing?  Wouldn't it be better to remove that cop than to have it break in some situations?
